devtools/replica-setup-pg/snapshot.sh

```diff
@@ -13,7 +13,13 @@ if [[ ! -f "$PG_DUMP" ]]; then
 fi
 
 # Read the file and match CREATE SCHEMA, CREATE TABLE and COPY statements
-SQLS=$(grep -Pzo 'CREATE SCHEMA [^;]+;\n|CREATE TABLE [^;]+;\n|COPY [^\\]+\\\.\n' "$PG_DUMP")
+SQLS=$(grep -Pzo 'CREATE DATABASE [^;]+;\n|\\connect [^\n]+\n|CREATE SCHEMA [^;]+;\n|CREATE TABLE [^;]+;\n|COPY [^\\]+\\\.\n' "$PG_DUMP")
+
+# Exclude "LOCALE_PROVIDER = '*' LOCALE = '*'" from CREATE DATABASE
+SQLS=$(echo "$SQLS" | sed 's/LOCALE_PROVIDER = [^ ;]*//g')
+SQLS=$(echo "$SQLS" | sed 's/LOCALE = [^ ;]*//g')
```

Comment:
`SQLS` can be very large, therefore this approach can be very slow. You can remove these unsupported options in `ConnectionHandler.convertQuery`.

Reply:
Yes, regex is ugly, but it's an acceptable implementation for our current situation. You can find an example of how doing so for MySQL protocol in 'backend.Handler.ComQuery'.